pkg/cmd/project/variables/create/create.go

```diff
@@ -0,0 +1,421 @@
+		fmt.Fprintf(opts.Out, "\nAutomation Command: %s\n", autoCmd)
+	}
+
+	return nil
+}
+
+func PromptMissing(opts *CreateOptions) error {
+	var project *projects.Project
+	var err error
+	if opts.Project.Value == "" {
+		project, err = projectSelector("You have not specified a Project. Please select one:", opts.GetAllProjectsCallback, opts.Ask)
+		if err != nil {
+			return nil
+		}
+		opts.Project.Value = project.GetName()
+	} else {
+		project, err = opts.GetProjectCallback(opts.Project.Value)
+		if err != nil {
+			return err
+		}
+	}
+
+	if opts.Name.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Name",
+			Help:    fmt.Sprintf("A name for this variable."),
+		}, &opts.Name.Value, survey.WithValidator(survey.ComposeValidators(
+			survey.MaxLength(200),
+			survey.MinLength(1),
+			survey.Required,
+		))); err != nil {
+			return err
+		}
+	}
+
+	question.AskDescription(opts.Ask, "", "Variable", &opts.Description.Value)
+
+	if opts.Type.Value == "" {
+		selectedType, err := selectors.SelectOptions(opts.Ask, "Select the type of the variable", getVariableTypeOptions)
+		if err != nil {
+			return err
+		}
+		opts.Type.Value = selectedType.Value
+	}
+
+	if !opts.IsPrompted.Value {
+		opts.Ask(&survey.Confirm{
+			Message: "Is this a prompted variable?",
+			Default: false,
+		}, &opts.IsPrompted.Value)
+	}
+
+	if opts.IsPrompted.Value {
+		if opts.PromptLabel.Value == "" {
+			if err := opts.Ask(&survey.Input{
+				Message: "Prompt Label",
+			}, &opts.PromptLabel.Value); err != nil {
+				return err
+			}
+		}
+
+		question.AskDescription(opts.Ask, "Prompt ", "Prompted Variable", &opts.PromptDescription.Value)
+
+		if opts.PromptType.Value == "" {
+			selectedPromptType, err := selectors.SelectOptions(opts.Ask, "Select the control type of the prompted variable", getControlTypeOptions)
+			if err != nil {
+				return err
+			}
+			opts.PromptType.Value = selectedPromptType.Value
+		}
+
+		if opts.PromptType.Value == PromptTypeDropdown && util.Empty(opts.PromptSelectOptions.Value) {
+			for {
+				var value string
+
+				if err := opts.Ask(&survey.Input{
+					Message: "Enter a selection option value (enter blank to end)",
+				}, &value, survey.WithValidator(survey.MaxLength(200))); err != nil {
+					return err
+				}
+
+				if value == "" {
+					break
+				}
+
+				var description string
+				if err := opts.Ask(&survey.Input{
+					Message: "Enter a selection option description",
+				}, &description, survey.WithValidator(survey.ComposeValidators(survey.Required))); err != nil {
+					return err
+				}
+
+				opts.PromptSelectOptions.Value = append(opts.PromptSelectOptions.Value, fmt.Sprintf("%s|%s", value, description))
+			}
+		}
+
+		if !opts.PromptRequired.Value {
+			opts.Ask(&survey.Confirm{
```

Comment:
very unlikely to happen but err assignment and check is missing here.

Reply:
good catch